File: plugins/dotnet/robocode.dotnet.napi/src/net/sf/robocode/api/Module.cs
using net.sf.robocode.serialization;
using robocode;

namespace net.sf.robocode.api
{
    public class Module
    {
        static Module()
        {
            RbSerializer.register(typeof (RobotStatus), 6);
            RbSerializer.register(typeof (BattleResults), 8);
            RbSerializer.register(typeof (Bullet), 9);
            RbSerializer.register(typeof (BattleEndedEvent), 0x20);
            RbSerializer.register(typeof (BulletHitBulletEvent), 0x21);
            RbSerializer.register(typeof (BulletHitEvent), 0x22);
            RbSerializer.register(typeof (BulletMissedEvent), 0x23);
            RbSerializer.register(typeof (DeathEvent), 0x24);
            RbSerializer.register(typeof (WinEvent), 0x25);
            RbSerializer.register(typeof (HitWallEvent), 0x26);
            RbSerializer.register(typeof (RobotDeathEvent), 0x27);
            RbSerializer.register(typeof (SkippedTurnEvent), 40);
            RbSerializer.register(typeof (ScannedRobotEvent), 0x29);
            RbSerializer.register(typeof (HitByBulletEvent), 0x2a);
            RbSerializer.register(typeof (HitRobotEvent), 0x2b);
            RbSerializer.register(typeof (KeyPressedEvent), 0x2c);
            RbSerializer.register(typeof (KeyReleasedEvent), 0x2d);
            RbSerializer.register(typeof (KeyTypedEvent), 0x2e);
            RbSerializer.register(typeof (MouseClickedEvent), 0x2f);
            RbSerializer.register(typeof (MouseDraggedEvent), 0x30);
            RbSerializer.register(typeof (MouseEnteredEvent), 0x31);
            RbSerializer.register(typeof (MouseExitedEvent), 50);
            RbSerializer.register(typeof (MouseMovedEvent), 0x33);
            RbSerializer.register(typeof (MousePressedEvent), 0x34);
            RbSerializer.register(typeof (MouseReleasedEvent), 0x35);
            RbSerializer.register(typeof (MouseWheelMovedEvent), 0x36);
        }

        public static void Init()
        {
        }
    }
}

File: robocode.api/src/main/java/net/sf/robocode/api/Module.java
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package net.sf.robocode.api;


import net.sf.robocode.core.IModule;
import net.sf.robocode.serialization.RbSerializer;
import robocode.*;

import java.util.List;


/**
 * @author Pavel Savara (original)
 */
public class Module implements IModule {
	public void afterLoaded(List<IModule> allModules) {
		RbSerializer.register(RobotStatus.class, RbSerializer.RobotStatus_TYPE);
		RbSerializer.register(BattleResults.class, RbSerializer.BattleResults_TYPE);
		RbSerializer.register(Bullet.class, RbSerializer.Bullet_TYPE);

		// events
		RbSerializer.register(RoundEndedEvent.class, RbSerializer.RoundEndedEvent_TYPE);
		RbSerializer.register(BattleEndedEvent.class, RbSerializer.BattleEndedEvent_TYPE);
		RbSerializer.register(BulletHitBulletEvent.class, RbSerializer.BulletHitBulletEvent_TYPE);
		RbSerializer.register(BulletHitEvent.class, RbSerializer.BulletHitEvent_TYPE);
		RbSerializer.register(BulletMissedEvent.class, RbSerializer.BulletMissedEvent_TYPE);
		RbSerializer.register(DeathEvent.class, RbSerializer.DeathEvent_TYPE);

		RbSerializer.register(WinEvent.class, RbSerializer.WinEvent_TYPE);
		RbSerializer.register(HitWallEvent.class, RbSerializer.HitWallEvent_TYPE);
		RbSerializer.register(RobotDeathEvent.class, RbSerializer.RobotDeathEvent_TYPE);
		RbSerializer.register(SkippedTurnEvent.class, RbSerializer.SkippedTurnEvent_TYPE);
		RbSerializer.register(ScannedRobotEvent.class, RbSerializer.ScannedRobotEvent_TYPE);
		RbSerializer.register(HitByBulletEvent.class, RbSerializer.HitByBulletEvent_TYPE);
		RbSerializer.register(HitRobotEvent.class, RbSerializer.HitRobotEvent_TYPE);
		RbSerializer.register(KeyPressedEvent.class, RbSerializer.KeyPressedEvent_TYPE);
		RbSerializer.register(KeyReleasedEvent.class, RbSerializer.KeyReleasedEvent_TYPE);
		RbSerializer.register(KeyTypedEvent.class, RbSerializer.KeyTypedEvent_TYPE);
		RbSerializer.register(MouseClickedEvent.class, RbSerializer.MouseClickedEvent_TYPE);
		RbSerializer.register(MouseDraggedEvent.class, RbSerializer.MouseDraggedEvent_TYPE);
		RbSerializer.register(MouseEnteredEvent.class, RbSerializer.MouseEnteredEvent_TYPE);
		RbSerializer.register(MouseExitedEvent.class, RbSerializer.MouseExitedEvent_TYPE);
		RbSerializer.register(MouseMovedEvent.class, RbSerializer.MouseMovedEvent_TYPE);
		RbSerializer.register(MousePressedEvent.class, RbSerializer.MousePressedEvent_TYPE);
		RbSerializer.register(MouseReleasedEvent.class, RbSerializer.MouseReleasedEvent_TYPE);
		RbSerializer.register(MouseWheelMovedEvent.class, RbSerializer.MouseWheelMovedEvent_TYPE);
	}
}
